multi_partner_learning.py

```diff
@@ -209,35 +198,43 @@ def compute_test_score(self):
 
         # After last epoch or if early stopping was triggered, evaluate model on the global testset
         logger.info("### Evaluating model on test data:")
-        model_evaluation = model_to_evaluate.evaluate(
-            x_test, y_test, batch_size=constants.DEFAULT_BATCH_SIZE, verbose=0)
+        model_evaluation_test_data= self.evaluate_model(model_to_evaluate, self.test_data)
         logger.info(f"   Model metrics names: {model_to_evaluate.metrics_names}")
-        logger.info(f"   Model metrics values: {['%.3f' % elem for elem in model_evaluation]}")
-        self.test_score = model_evaluation[1]  # 0 is for the loss
+        logger.info(f"   Model metrics values: {['%.3f' % elem for elem in model_evaluation_test_data]}")
+        self.test_score = model_evaluation_test_data[1]  # 0 is for the loss
         self.nb_epochs_done = self.epoch_index + 1
 
         # Plot training history # TODO: move the data saving and plotting in dedicated functions
         if self.is_save_data:
             self.save_data()
-        
+
         self.save_final_model_weights(model_to_evaluate)
 
         logger.info("Training and evaluation on multiple partners: done.")
         end = timer()
         self.learning_computation_time = end - start
-        
+
     def save_final_model_weights(self, model_to_save):
         """Save final model weights"""
-        
+
         model_folder = os.path.join(self.save_folder, 'model')
 
         if not os.path.isdir(model_folder):
             os.makedirs(model_folder)
-            
-        model_to_save.save_weights(os.path.join(model_folder, self.dataset_name+'_final_weights.h5'))
-        model_weights = model_to_save.get_weights()
-            
-        np.save(os.path.join(model_folder, self.dataset_name+'_final_weights.npy'), 
+
+        if isinstance(model_to_save, type(LogisticRegression())):
+            dump(model_to_save, os.path.join(model_folder, self.dataset_name+'_final_weights.h5'))
+            coefs = np.array(model_to_save.coef_ )
+            intercepts = np.array(model_to_save.intercept_ )
+
+            np.savez(os.path.join(model_folder, self.dataset_name+'_final_weights.npy'),
+                name1=coefs, name2=intercepts)
```

Comment:
Hmmm, the `savez` seems like a French verb? :laughing:

Reply:
When testing I got `TypeError: _save_dispatcher() got an unexpected keyword argument 'name1'`

Reply:
My mistake of course 🤦‍♂️